cypress/e2e/29-templates.cy.ts

```diff
@@ -29,12 +29,33 @@ describe('Templates', () => {
 		cy.url().then(($url) => {
 			expect($url).to.include('/workflow/new?templateId=1234');
 		});
-		
+
 		workflowPage.getters.canvasNodes().should('have.length', 4);
 		workflowPage.getters.stickies().should('have.length', 1);
 		workflowPage.actions.shouldHaveWorkflowName(OnboardingWorkflow.name);
 	});
 
+	it('should save template id with the workflow', () => {
+		cy.intercept('POST', '/rest/workflows').as('saveWorkflow');
+		cy.visit(templatesPage.url);
+		templatesPage.getters.firstTemplateCard().click();
+		cy.url().should('include', '/templates/');
+
+		cy.url().then(($url) => {
+			const templateId = $url.split('/').pop();
+
+			templatesPage.getters.useTemplateButton().click();
+			cy.url().should('include', '/workflow/new');
+			workflowPage.actions.saveWorkflowOnButtonClick();
+
+			cy.wait('@saveWorkflow').then((interception) => {
+				expect(interception.request.body.meta.templateId).to.equal(templateId);
```

Comment:
is this really a good e2e test? seems to be more of an integration test. it's too coupled to the api request and does not tell you it was stored or used correctly.. Could you think of a better approach, even though it does not seem we are using this yet.. maybe check the json of loaded workflow or when copying/pasting?

Reply:
Good point on the API coupling. The problem with meta is that it has to rely for something under the hood (like you mentioned). Goal here is to test if template user chooses is correctly saved, but I guess we can check that also by re-opening the workflow, will try something like that.

Reply:
ya this might be one of those scenarios where a unit test is better.. but nice solution here..